enclave/core/sgx/random_internal.c

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/string.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/rdrand.h>
+
+// The RDRAND generats 8-byte random value.
+#define RDRAND_BYTES 8
+
+oe_result_t oe_random_internal(void* data, size_t size)
+{
+    for (size_t i = 0; i < size; i += RDRAND_BYTES)
+    {
+        size_t request_size = size - i;
+        if (request_size > RDRAND_BYTES)
+        {
+            request_size = RDRAND_BYTES;
+        }
+        uint64_t random_bytes = oe_rdrand();
```

Comment:
The previous implementation of oe_random_internal called `mbedtls_ctr_drbg_random`

That function has a more complex implementation: https://github.com/openenclave/openenclave/blob/040beb10dba521ee95af10b0898707e61cceb3b6/3rdparty/mbedtls/mbedtls/library/ctr_drbg.c#L444

Is there any cryptographic consequence of using one implementation vs the other?

Reply:
@anakrish Yes, but that is the intent of #892: to provide a public enclave API that gets access to trusted HW randomness that is not mediated by the enclave runtime. Internally, the oeenclave functionality still uses the mbedtls ctr_drbg.